basiclibrary/src/test/java/basiclibrary/LibraryTest.java

```diff
@@ -7,9 +7,40 @@
 import static org.junit.Assert.*;
 
 public class LibraryTest {
-    @Test public void testSomeLibraryMethod() {
+    @Test public void testRollMethod() {
         int number = 4;
         int[] result = Library.roll(number);
         assertEquals(number, result.length);
     }
+
+    @Test public void testContainsDuplicatesFalseMethod() {
+        int[] inputArr = {2,4,5,6,8};
+        boolean actual = Library.containsDuplicates(inputArr);
+        assertFalse(actual);
+    }
+
+    @Test public void testContainsDuplicatesTrueMethod() {
+        int[] inputArr = {2,4,5,5,8};
+        boolean actual = Library.containsDuplicates(inputArr);
+        assertTrue(actual);
+    }
+
+    @Test public void testcalCulatingAveragesMethod() {
+        int[] inputArr = {2,4,5,6,8};
+        int actual = Library.calculatingAverages(inputArr);
+        int expected = 5;
```

Comment:
This test is failing for a few reasons. First the average for `expected` should be 4.666. Also, since the method `calculatingAverages` returns a float, you would want the type of `expected` to also be float. When you test a float you will want to use something like: `assertEquals(expected, actual, 0.01);` The last number is delta, which is the margin of error.

Reply:
Thanks! As I followed Nicholas code review, I forgot to update that part!